ops/testing.py

```diff
@@ -221,6 +225,21 @@ def set_can_connect(self, container: Union[str, model.Container], val: bool):
             container = self.model.unit.get_container(container)
         self._backend._set_can_connect(container._pebble, val)
 
+    def get_container_filesystem(
+        self, container: Union[str, model.Container]
+    ) -> "ContainerFilesystem":
+        """Return an interface for conducting filesystem operations on the specified container.
+
+        This method returns an object which provides an interface for filesystem-related
+        operations (push, pull, and so on) on the specified container. The operations available
+        through this interface are always permitted and unaffected by the
+        :meth:`.set_can_connect` method.
+        """
+        if isinstance(container, str):
+            container = self.model.unit.get_container(container)
+        proxy = _ContainerFilesystemProxy(container=container, backend=self._backend)
+        return cast(ContainerFilesystem, proxy)
```

Comment:
I think `return cast(ContainerFilesystem, proxy)` and `return cast(model.Container, proxy)` doesn't have any difference. Since from an outsider's perspective, the return type is `ContainerFilesystem`. The cast here is just make the type checker happy.

Reply:
@PietroPasotti Having the return type as `model.Container` and avoiding the protocol would work, but I think it sends the wrong signal to the user -- we only want them to use this to interact with the file system (hence the name `get_container_filesystem`). The idea is to show them it's special access for use only in tests, not a real container which has can-connect behaviour.